cmake/modules/libbpf.cmake

```diff
@@ -21,12 +21,12 @@ else()
     ExternalProject_Add(
         libbpf
         PREFIX "${PROJECT_BINARY_DIR}/libbpf-prefix"
-        DEPENDS zlib
+        DEPENDS zlib libelf
         URL "https://github.com/libbpf/libbpf/archive/refs/tags/v1.0.1.tar.gz"
         URL_HASH
         "SHA256=3d6afde67682c909e341bf194678a8969f17628705af25f900d5f68bd299cb03"
         CONFIGURE_COMMAND mkdir -p build root
-        BUILD_COMMAND BUILD_STATIC_ONLY=y OBJDIR=${LIBBPF_BUILD_DIR}/build DESTDIR=${LIBBPF_BUILD_DIR}/root make -C ${LIBBPF_SRC}/libbpf/src install install_uapi_headers
+        BUILD_COMMAND ${CMD_MAKE} BUILD_STATIC_ONLY=y OBJDIR=${LIBBPF_BUILD_DIR}/build DESTDIR=${LIBBPF_BUILD_DIR}/root NO_PKG_CONFIG=1 "EXTRA_CFLAGS=-I${LIBELF_INCLUDE} -I${ZLIB_INCLUDE}" "LDFLAGS=-Wl,-Bstatic" "EXTRA_LDFLAGS=-L${LIBELF_SRC}/libelf/libelf -L${ZLIB_SRC}" -C ${LIBBPF_SRC}/libbpf/src install install_uapi_headers
```

Comment:
Properly link against our own built libelf and zlib. Avoid using `pkg_config` to find the correct libraries.